assembly/buffer/index.ts

```diff
@@ -287,6 +287,33 @@ export class Buffer extends Uint8Array {
 }
 
 export namespace Buffer {
+  export namespace ASCII {
+    export function encode(str: string): ArrayBuffer {
+      let length = str.length;
+      let output = __alloc(length, idof<ArrayBuffer>());
+      for (let i = 0; i < length; i++) {
+        let char = load<u16>(changetype<usize>(str) + <usize>(i << 1));
+        store<u8>(output + <usize>i, <u8>(0b01111111 & char));
```

Comment:
Could be simplify to
```suggestion
        store<u8>(output + <usize>i, char & 0x7F);
```